src/WebAPI/Routes/ConversationHistoryRoutes.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Application.Contracts;
+using Application.Services;
+
+namespace WebAPI.Routes;
+
+public static class ConversationHistoryRoutes
+{
+    private const string ChatHistoryPolicy = nameof(ChatHistoryPolicy);
+    
+    public static void MapConversationRoutes(this WebApplication app)
+    {
+        var routeGroup = app.MapGroup("").WithTags("ConversationHistory").WithOpenApi();
+        
+        routeGroup
+            .MapGet(
+                "/ConversationById",
```

Comment:
Id should be coming from the route. not the query string.

Consider changing the route to "/Conversations/{id}"